index.js

```diff
@@ -154,17 +154,42 @@ export default class ClearvisioAppointmentBooker {
   }
 
   async loadStore(storeCode) {
-    var stores = await api.get(this.store, `stores?code=${storeCode}`);
-    this.store.dispatch('store/set', stores[0]);
+    try {
+      var stores = await api.get(this.store, `stores?code=${storeCode}`);
+      if ( stores[0] == undefined) {
+        this.store.dispatch('moduleState/set', 'error.storeCode');
+        return;
+      }
+      this.store.dispatch('store/set', stores[0]);
+    } catch (error) {
+      if (error.code == 403) {
+        this.store.dispatch('moduleState/set', 'error.403');
+      } else {
+        this.store.dispatch('moduleState/set', 'error.storeCode');
+      }
+    }
   }
 
   async loadEyeExaminationProcesses({eyeExaminationProcessId}) {
     if (eyeExaminationProcessId) {
-      return [await api.get(this.store, `eye_examination_processes/${eyeExaminationProcessId}`)]
-        .filter((process) => process);
+      try {
+        const  process = await api.get(this.store, `eye_examination_processes/${eyeExaminationProcessId}`);
+        return process
+      } catch(error) {
+        if (error.code == 404){
+          this.store.dispatch('moduleState/set', 'error.missingConfiguredProcessId');
+          return;
+        } 
+      }
     }
     var storeEntity = this.store.get().store;
-    return await api.get(this.store, `eye_examination_processes?hasLength&chain=${storeEntity.chain['@id']}`);
+    var examination = await api.get(this.store, `eye_examination_processes?hasLength&chain=${storeEntity.chain['@id']}`);
+    if (examination.length == 0) {
+      this.store.dispatch('moduleState/set', 'error.noLenght');
+      return;
+    } else{
```

Comment:
ha van egy if-em, amiben returnolok/throwolok (vagy másutt brake-elek, continue-olok) akkor az else ágot nem szoktuk mint else ki írni, mert minek. ha feltétel teljesül bementünk az if ágba és aztán kiléptünk valahogy. Tehát a másik ág az maradhat a "normál ág", kint minden extra kapcsoszárójelezéstől és behúzástól. És akkor annyival is egyszerübb a kódod.